ichnaea/locate/searcher.py

```diff
@@ -27,7 +27,8 @@ class Searcher(StatsLogger):
 
     provider_classes = ()
 
-    def __init__(self, session_db, geoip_db, settings, *args, **kwargs):
+    def __init__(self, session_db, geoip_db, settings,
+                 redis_client=None, *args, **kwargs):
```

Comment:
I'd move redis_client before the settings argument and make it required. It's more similar to session and geoip as some sort of external connection object.